src/EnergyPlus/HVACInterfaceManager.cc

```diff
@@ -343,6 +410,30 @@ void UpdateHVACInterface(EnergyPlusData &state,
         if (airLoopConv.HVACPressueSupplyDeck2ToDemandTolValue[0] > DataConvergParams::HVACPressToler) {
             OutOfToleranceFlag = true; // Something has changed--resimulate the other side of the loop
         }
+
+        if (state.dataContaminantBalance->Contaminant.CO2Simulation) {
+            TmpRealARR = airLoopConv.HVACCO2SupplyDeck2ToDemandTolValue;
+            airLoopConv.HVACCO2SupplyDeck2ToDemandTolValue[0] = std::abs(state.dataLoopNodes->Node(OutletNode).CO2 - thisInletNode.CO2);
+            for (int logIndex = 1; logIndex < DataConvergParams::ConvergLogStackDepth; logIndex++) {
+                airLoopConv.HVACCO2SupplyDeck2ToDemandTolValue[logIndex] = TmpRealARR[logIndex - 1];
+            }
+            if (airLoopConv.HVACCO2SupplyDeck2ToDemandTolValue[0] > DataConvergParams::HVACCO2Toler) {
+                airLoopConv.HVACCO2NotConverged[2] = true;
+                OutOfToleranceFlag = true; // Something has changed--resimulate the other side of the loop
+            }
+        }
+        if (state.dataContaminantBalance->Contaminant.GenericContamSimulation) {
+            TmpRealARR = airLoopConv.HVACGenContamSupplyDeck2ToDemandTolValue;
+            airLoopConv.HVACGenContamSupplyDeck2ToDemandTolValue[0] =
+                std::abs(state.dataLoopNodes->Node(OutletNode).GenContam - thisInletNode.GenContam);
+            for (int logIndex = 1; logIndex < DataConvergParams::ConvergLogStackDepth; logIndex++) {
+                airLoopConv.HVACGenContamSupplyDeck2ToDemandTolValue[logIndex] = TmpRealARR[logIndex - 1];
+            }
+            if (airLoopConv.HVACGenContamSupplyDeck2ToDemandTolValue[0] > DataConvergParams::HVACGenContamToler) {
+                airLoopConv.HVACGenContamNotConverged[2] = true;
+                OutOfToleranceFlag = true; // Something has changed--resimulate the other side of the loop
+            }
```

Comment:
Yeah, checking the value of variable [0] and then saying converged[2] = true.  This is not easy to read and understand.

Reply:
@Myoldmopar  a new constexpr variable suggested by you may address your concern.

Reply:
So this DemandTolValue[0] is not related to the same indexing that the converged[2] was referencing?  It doesn't have to be changed now, and using CalledFrom is helpful indeed.